### DISCUSSION TITLE
Home Assistant OS 12.3

### DISCUSSION THREAD

**Discussion by agners** [home-assistant/operating-system#3345] [Announcements]
## Home Assistant Operating System

* Prepare 12.3.rc1 (#3327) @sairon
* Bump buildroot to update BlueZ to 5.75 (#3326) @sairon
* Bump buildroot to update linux-pam (#3325) @sairon
* Update instructions for ODROID M1 Petitboot update (#3317) @salim-b

## Build System

* Enable KVM for running tests in GH Actions (#3329) @sairon

## Raspberry Pi

* RaspberryPi: Update kernel to 6.6.28 - stable_20240423 (#3336) @sairon
* Fix incorrect mapping of DT overlays on RPi 5 (#3335) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon
* Increase maximum length of usb_storage.quirks in RPi kernel params (#3311) @sairon
* Update Raspberry Pi 3 device tree list (#3301) @agners

## Home Assistant Yellow

* RaspberryPi: Update kernel to 6.6.28 - stable_20240423 (#3336) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon
* Increase maximum length of usb_storage.quirks in RPi kernel params (#3311) @sairon

## Home Assistant Green

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon

## Open Virtual Appliance

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon

## Generic x86-64

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon
* Revert GRUB2 patch causing boot failure on some old x86_64 BIOSes (#3324) @sairon

## Hardkernel ODROID

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon
* ODROID-N2/C2/C4: Force 40MHz instead of 24MHz as eMMC clock (#3319) @agners

## ASUS Tinker

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon

## Khadas VIM Series

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon

## Generic aarch64

* Linux: Update kernel to 6.6.29 (#3334) @sairon
* Enable kernel userspace crypto API interface for all platforms (#3330) @sairon

<hr /><em>This discussion was created from the release <a href='https://github.com/home-assistant/operating-system/releases/tag/12.3'>Home Assistant OS 12.3</a>.</em>

**Comment by decibelio99** [comment]
Hello, the first thing is to thank you for your great work.
I have problems with this version on my odroid n2.
Yesterday when I updated it didn't start, then I tried to install an image of version 12.3 and it didn't work either. I had to install version 11.5 and that works. All the best

**Reply by sairon** [reply]
If you're running the system from an eMMC, then you're most likely affected by this bug: https://github.com/home-assistant/operating-system/issues/3351

Could you please also share details about your eMMC module there?

**Reply by decibelio99** [reply]
Yes, I have a 16GB eMMC, I don't know how to find the details. It is original from Odroid. I'm testing 12.2

**Comment by laojr1983** [comment]
I installed in generic x86-64 and didn't work. Connection refused in 8123 port (home assistant) and ok in 4357 port (supervisor, all item oks in supervisor).

**What is the problem?**

Note: I tried using "IP" and "homeassistant.local". The problem is the same.

**Comment by david20100** [comment]
Bonjour
RPI3 impossible de me connecter en local port 8123, je remet une ancienne version

**Reply by laojr1983** [reply]
You need to format the disk in mode ntfs, using linux (Ubuntu, example) to do it. This was my problem.

**Comment by ROnderstal** [comment]
Installed this update and now keeps booting into 

slot A ok=1 try=3
slot B ok=1 try=3
Slot A rescue shell
Slot B rescue shell

And if I am lucky one of them will work.

HP T620 Quad Core
32GB sata1 sandisksdsa6mm-032G-1006

**Comment by yarikoptic** [comment]
Is there a chance to get checksums along with the images?  I am asking because in attempt to fix up hassio sdcard, I would like to extract individual squashfs images from the .img.gz

<details>
<summary>but whenever 3rd partition is coming out ok from haos_rpi3-64-12.3.img.xz, the 5th partition is coming out not looking like squashfs :-/ </summary> 

```shell
❯ sha256sum haos_rpi3-64-12.3.img.xz > haos_rpi3-64-12.3.img.xz.sha256sum
❯ cat haos_rpi3-64-12.3.img.xz.sha256sum
d96f8c9a0f9329dc39b403b1f1737950af9377c395bff126d91d77e8178102b3  haos_rpi3-64-12.3.img.xz

❯ xz --decompress haos_rpi3-64-12.3.img.xz
xz --decompress haos_rpi3-64-12.3.img.xz  31.34s user 0.60s system 99% cpu 31.944 total
❯ ls
haos_rpi3-64-12.3.img  haos_rpi3-64-12.3.img.xz.sha256sum  haos_rpi3-64-12.3.raucb  tmp/

❯ fdisk -l haos_rpi3-64-12.3.img
Disk haos_rpi3-64-12.3.img: 2 GiB, 2147483648 bytes, 4194304 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes
Disklabel type: gpt
Disk identifier: 117224D8-16BB-4329-AD88-7D30B3C2300E

Device                   Start     End Sectors  Size Type
haos_rpi3-64-12.3.img1    2048   67583   65536   32M Microsoft reserved
haos_rpi3-64-12.3.img2   67584  116735   49152   24M Linux filesystem
haos_rpi3-64-12.3.img3  116736  641023  524288  256M Linux filesystem
haos_rpi3-64-12.3.img4  641024  690175   49152   24M Linux filesystem
haos_rpi3-64-12.3.img5  690176 1214463  524288  256M Linux filesystem
haos_rpi3-64-12.3.img6 1214464 1230847   16384    8M Linux filesystem
haos_rpi3-64-12.3.img7 1230848 1427455  196608   96M Linux filesystem
haos_rpi3-64-12.3.img8 1427456 4048895 2621440  1.3G Linux filesystem

❯ dd if=haos_rpi3-64-12.3.img of=haos_rpi3-64-12.3.img3 bs=512 skip=116736 count=524288
524288+0 records in
524288+0 records out
268435456 bytes (268 MB, 256 MiB) copied, 1.55595 s, 173 MB/s
❯ file haos_rpi3-64-12.3.img
haos_rpi3-64-12.3.img: DOS/MBR boot sector; partition 1 : ID=0xc, active, start-CHS (0x0,32,33), end-CHS (0x4,52,48), startsector 2048, 65536 sectors; partition 2 : ID=0xee, start-CHS (0x0,0,2), end-CHS (0x0,32,32), startsector 1, 2047 sectors

❯ dd if=haos_rpi3-64-12.3.img of=haos_rpi3-64-12.3.img5 bs=512 skip=690176 count=524288
524288+0 records in
524288+0 records out
268435456 bytes (268 MB, 256 MiB) copied, 1.6296 s, 165 MB/s
❯ file haos_rpi3-64-12.3.img5
haos_rpi3-64-12.3.img5: data
```
</details>

and is really just full of 0s:

```shell
❯ hexdump -C haos_rpi3-64-12.3.img5
00000000  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
*
10000000
```

or should they be such in a "fresh installation"?

**Reply by yarikoptic** [reply]
probably nevermind -- I guess that is a special setup of hassio to be able to do near atomic upgrades -- write to "the other" squashfs and then flip from which to go ...  anyways -- reflashed all first 6 filesystems using https://gist.github.com/yarikoptic/e5a075ecb8ee5e7fc2c85637408b4d2e and all is good now.
